src/dsp_tools/commands/xmlupload/check_consistency_with_ontology.py

```diff
@@ -0,0 +1,123 @@
+    If it finds any invalid properties or classes, they are printed out and a UserError is raised.
+
+     Args:
+         onto_client: client for the ontology retrieval
+         root: root of the XML
+
+     Raises:
+         UserError: if there are any invalid properties or classes
+    """
+    onto_check_info = OntoCheckInformation(
+        default_ontology_prefix=onto_client.default_ontology,
+        onto_lookup=onto_client.get_all_ontologies_from_server(),
+        save_location=onto_client.save_location,
+    )
+    classes, properties = _get_all_classes_and_properties(root)
+    _find_problems_in_classes_and_properties(classes, properties, onto_check_info)
+
+
+def _get_all_classes_and_properties(root: etree._Element) -> tuple[list[list[str]], list[list[str]]]:
+    classes = _get_all_class_types_and_ids(root)
+    properties = []
+    for resource in root.iterchildren(tag="resource"):
+        properties.extend(_get_all_property_names_and_resource_ids(resource))
+    return classes, properties
+
+
+def _get_all_class_types_and_ids(root: etree._Element) -> list[list[str]]:
+    return [[resource.attrib["id"], resource.attrib["restype"]] for resource in root.iterchildren(tag="resource")]
+
+
+def _get_all_property_names_and_resource_ids(resource: etree._Element) -> list[list[str]]:
+    return [[resource.attrib["id"], prop.attrib["name"]] for prop in resource.iterchildren() if prop.tag != "bitstream"]
+
+
+def _find_problems_in_classes_and_properties(
+    classes: list[list[str]], properties: list[list[str]], onto_check_info: OntoCheckInformation
+) -> None:
+    class_problems = _diagnose_all_classes(classes, onto_check_info)
+    property_problems = _diagnose_all_properties(properties, onto_check_info)
+    if not class_problems and not property_problems:
+        return None
+    problems = InvalidOntologyElements(
+        save_path=onto_check_info.save_location, classes=class_problems, properties=property_problems
+    )
+    problems.execute_problem_protocol()
+
+
+def _diagnose_all_classes(classes: list[list[str]], onto_check_info: OntoCheckInformation) -> list[list[str]]:
+    problem_list = []
+    for cls_info in classes:
+        if problem := _diagnose_class(cls_info, onto_check_info):
+            problem_list.append(problem)
+    return problem_list
+
+
+def _diagnose_class(class_info: list[str], onto_check_info: OntoCheckInformation) -> list[str] | None:
+    try:
+        prefix, cls_ = _get_prefix_and_prop_cls_identifier(class_info[1], onto_check_info.default_ontology_prefix)
+    except BaseError:
+        class_info.append("Resource type does not follow a known ontology pattern")
+        return class_info
+    try:
+        if cls_ not in onto_check_info.onto_lookup[prefix].classes:
+            class_info.append("Invalid Class Type")
+            return class_info
+        return None
+    except KeyError:
+        class_info.append("Unknown ontology prefix")
+        return class_info
```

Comment:
I personally don't like the over-use of exceptions, and all of this could be done by if-checking (and the performance difference should be negligible, even in large projects)

Reply:
I understand your point. I tend to do them not only for performance but also if I consider it to be rare. I will have a think about it.

Reply:
For me, only readability counts in this matter. I dislike too many try-exepts, no matter how often they are executed. For me the question is rather: How often do I have to read it? The question "How often is it executed?" doesn't matter for me.

Reply:
Right, and in that case, I as opposed to you often times I prefer Errors...